scripts/injectGlobalMixins.ts

```diff
@@ -0,0 +1,103 @@
+import { getPackages } from '@lerna/project';
+import batchPackages from '@lerna/batch-packages';
+import filterPackages from '@lerna/filter-packages';
+import path from 'path';
+import fs from 'fs';
+
+/**
+ * simplified interface for a package.json
+ */
+interface SimplePackageJson
+{
+    location: string
+    name: string
+    dependencies: Record<string, string>
+}
+
+/**
+ * Gets all the non-private packages package.json data
+ */
+async function getSortedPackages(): Promise<SimplePackageJson[]>
+{
+    const packages = await getPackages(process.cwd());
+
+    const filtered = filterPackages(packages, undefined, undefined, false);
+
+    return batchPackages(filtered)
+        .reduce((arr: SimplePackageJson[], batch: SimplePackageJson[]) => arr.concat(batch), []);
+}
+
+/**
+ * Adds global reference to the start of a packages `index.d.ts` file
+ */
+function writeToIndex(basePath: string, dataToWrite: string): void
+{
+    const indexDtsPath = path.resolve(basePath, './index.d.ts');
+    const file = fs.readFileSync(indexDtsPath, { encoding: 'utf8' }).toString().split('\n');
+
+    file.unshift(dataToWrite);
+    fs.writeFileSync(indexDtsPath, file.join('\n'));
+}
+
+/**
+ * This is a workaround for https://github.com/pixijs/pixi.js/issues/6993
+ *
+ * All this script does is inject a path reference into a packages `index.d.ts` if a `global.d.ts`
+ * exists.
+ */
+async function start(): Promise<void>
+{
+    let pixiLocation: string;
+    let pixiLegacyLocation: string;
+    let pixiGlobalMixins = '';
+    let pixiLegacyGlobalMixins = '';
+
+    const packages = await getSortedPackages();
+    const legacyPackages = Object.keys(packages.find((pkg) => pkg.name === 'pixi.js-legacy').dependencies);
+
+    packages.forEach((pkg) =>
+    {
+        const basePath = path.relative(process.cwd(), pkg.location);
+
+        if (pkg.name === 'pixi.js')
+        {
+            pixiLocation = pkg.location;
+
+            return;
+        }
+        if (pkg.name === 'pixi.js-legacy')
+        {
+            pixiLegacyLocation = pkg.location;
+
+            return;
+        }
+
+        const globalDtsPath = path.resolve(basePath, './global.d.ts');
+
+        if (fs.existsSync(globalDtsPath))
+        {
+            const pixiTypeData = `/// <reference path="../${pkg.name}/global.d.ts" />\n`;
```

Comment:
Can you see if you can use `/// <reference types="${pkg.name}/global.d.ts" />`? It would be nice to avoid any relative pathing issues (for instance symlinked packages) and using node's resolve instead. Not sure if this works tho.

Reply:
ahhhh!!

this didn't work, however I have found this https://www.typescriptlang.org/docs/handbook/triple-slash-directives.html

so we can do `/// <reference types="@pixi/core" />` inside `pixi.js`

Reply:
@bigtimebuddy updated!